src/cvodes/cvodes.c

```diff
@@ -9916,71 +9904,32 @@ static int cvQuadSensRhs1InternalDQ(CVodeMem cv_mem, int is, sunrealtype t,
  * -----------------------------------------------------------------
  */
 
-/*
- * cvProcessError is a high level error handling function.
- * - If cv_mem==NULL it prints the error message to stderr.
- * - Otherwise, it sets up and calls the error handling function
- *   pointed to by cv_ehfun.
- */
-
-void cvProcessError(CVodeMem cv_mem, int error_code, const char* module,
-                    const char* fname, const char* msgfmt, ...)
+void cvProcessError(CVodeMem cv_mem, int error_code, int line, const char* func,
+                    const char* file, const char* msgfmt, ...)
 {
-  va_list ap;
-  char msg[256];
-
   /* Initialize the argument pointer variable
      (msgfmt is the last required argument to cvProcessError) */
-
+  va_list ap;
   va_start(ap, msgfmt);
 
   /* Compose the message */
-
-  vsprintf(msg, msgfmt, ap);
+  size_t msglen = vsnprintf(NULL, 0, msgfmt, ap) + 1;
+  char* msg     = (char*)malloc(msglen);
+  vsnprintf(msg, msglen, msgfmt, ap);
 
   if (cv_mem == NULL)
-  { /* We write to stderr */
-#ifndef NO_FPRINTF_OUTPUT
-    fprintf(stderr, "\n[%s ERROR]  %s\n  ", module, fname);
-    fprintf(stderr, "%s\n\n", msg);
-#endif
+  {
+    SUNGlobalFallbackErrHandler(line, func, file, msgfmt, error_code, ap);
   }
   else
-  { /* We can call ehfun */
-    cv_mem->cv_ehfun(error_code, module, fname, msg, cv_mem->cv_eh_data);
+  {
+    /* Call the SUNDIALS main error handler */
+    SUNHandleErrWithMsg(line, func, file, msg, error_code, cv_mem->cv_sunctx);
   }
 
   /* Finalize argument processing */
   va_end(ap);
-
-  return;
-}
-
-/*
- * cvErrHandler is the default error handling function.
- * It sends the error message to the stream pointed to by cv_errfp.
- */
-
-void cvErrHandler(int error_code, const char* module, const char* function,
-                  char* msg, void* data)
-{
-  CVodeMem cv_mem;
-  char err_type[10];
-
-  /* data points to cv_mem here */
-
-  cv_mem = (CVodeMem)data;
-
-  if (error_code == CV_WARNING) { sprintf(err_type, "WARNING"); }
```

Comment:
There are couple uses of `cvProcessError` where the message is intended to be a warning rather than an error.